pkgs/development/coq-modules/ssprove/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, mkCoqDerivation, coq, version ? null
+, equations
+, mathcomp
+, mathcomp-analysis
+, extructures
+, deriving
+}:
+
+(mkCoqDerivation {
+  pname = "ssprove";
+  owner = "SSProve";
+
+  inherit version;
+  defaultVersion = with lib.versions; lib.switch [coq.coq-version mathcomp.version] [
+    { cases = ["8.19" "2.2.0" ]; out = "0.2.0"; }
+    { cases = ["8.18" "2.1.0" ]; out = "0.2.0"; }
```

Comment:
```suggestion
    { cases = [(range "8.18" "8.19") (isGe "2.1") ]; out = "0.2.0"; }
```

Reply:
I'm not a big fan of the `isGe` option because I do not know whether SSProve is compatible with the next verison of mathcomp. For the version that I listed, I have CI jobs that attest that.

Would the mathcomp guys take care of the compatibility issues when the upgrade? Then I would be fine with this change.

Reply:
What about using `range "2.1" "2.2"`?